lib/active_merchant/billing/gateways/redsys.rb

```diff
@@ -531,7 +533,7 @@ def build_merchant_mpi_external(xml, data)
         ds_merchant_mpi_external[:dsTransID] = data[:dsTransID] if data[:dsTransID]
         ds_merchant_mpi_external[:authenticacionValue] = data[:authenticacionValue] if data[:authenticacionValue]
         ds_merchant_mpi_external[:protocolVersion] = data[:protocolVersion] if data[:protocolVersion]
-
+        ds_merchant_mpi_external[:Eci] = data[:eci] if data[:eci]
```

Comment:
This seems to have a duplicate line a few lines up; is this intentional? 

https://github.com/activemerchant/active_merchant/pull/3855/files#diff-c1465fbdcc1b656226ad4ffdda93044a9b05262de9adce1f7a16d70f1fbdc435R530

Looks like the keys have different casing.

Reply:
That's correct the only difference is casting, but I added it intentionally since he documentation states ECI for v1 and Eci for v2.

Reply:
If `data[:eci]` is present here it will be added twice under both casing types, which isn't ideal. What do you think about setting `eci_v1` or `eci_v2` in `add_external_mpi_fields` and then setting the proper casing here accordingly instead?

Reply:
Great suggestion. I'll make that change